lib/DDG/Spice/Cryptocurrency.pm

```diff
@@ -0,0 +1,174 @@
+package DDG::Spice::Cryptocurrency;
+# ABSTRACT: Cryptocurrency converter and exchange rate lookup provided by cryptonator.com
+# Borrows from DDG::Spice:Currency
+ 
+use DDG::Spice;
+with 'DDG::SpiceRole::NumberStyler';
+
+name "Cryptocurrency";
+source "cryptonator.com";
+icon_url "https://www.cryptonator.com/ui/img/favicon.png";
+description "Cryptocurrency converter and exchange rate lookup provided by cryptonator.com";
+primary_example_queries "convert ltc to btc", "50 ftc to btc";
+secondary_example_queries "50 ltc";
+category "conversions";
+topics "economy_and_finance";
+code_url "https://github.com/claytonspinner/zeroclickinfo-spice/blob/master/lib/DDG/Spice/Cryptocurrency.pm";
+attribution web => ['http://cryptonator.com','cryptonator.com'],
+            github => ["https://github.com/claytonspinner", "Clayton Spinner"],
+            email => ["clayton.spinner\@gmail.com", "Clayton Spinner"];
+
+# Get all the valid currencies from a text file.
+my @currTriggers;
+my @currencies = share('cryptocurrencylist.txt')->slurp;
+my %currHash = ();
+
+foreach my $currency (@currencies){
+    chomp($currency);
+    my @currency = split(/,/,$currency);
+    push(@currTriggers, @currency);
+    $currHash{$currency[0]} = \@currency;
+}
+
+# Used when a single currency is given in the query.
+# These currencies should be handled by DDG::Spice::Currency or DDG::Spice::Bitcoin.
+my @excludedCurrencies = (
+    'btc',
+    'aud',
+    'cny',
+    'eur',
+    'gbp',
+    'hkd',
+    'jpy',
+    'nzd',
+    'pln',
+    'rur',
+    'sgd',
+    'usd'
+);
+
+#Define regexes
+my $currency_qr = join('|', @currTriggers);
+my $question_prefix = qr/(?:convert|what (?:is|are|does)|how (?:much|many) (?:is|are))?\s?/;
+my $rate_qr = qr/\s(?:rate|exchange|exchange rate|conversion)/i;
+my $into_qr = qr/\s(?:en|in|to|in ?to|to|from)\s/i;
+my $vs_qr = qr/\sv(?:ersu|)s\.?\s/i;
+my $number_re = number_style_regex();
+
+my $guard = qr/^$question_prefix($number_re*?)\s?($currency_qr)(?:s)?(?:$into_qr|$vs_qr|$rate_qr|\s)?($number_re*?)\s?($currency_qr)?(?:s)?\??$/i;
+
+# http://www.cryptonator.com/api/secondaries?primary=BTC
+# http://www.cryptonator.com/api/ticker/ltc-ftc
+
+triggers query_lc => qr/$currency_qr/;
+
+spice from => '([^/]+)/([^/]+)/([^/]*)';
+spice to => 'http://www.cryptonator.com/api/$1/$2?primary=$3';
```

Comment:
Please change http to https--it redirects, and we don't allow the endpoints to redirect internally.